modular_zubbers/code/modules/clothing/neck/collars.dm

```diff
@@ -9,3 +9,75 @@
 /obj/item/clothing/neck/human_petcollar/locked/ringbell/Initialize(mapload)
 	. = ..()
 	AddComponent(/datum/component/squeak, list('modular_zubbers/sound/misc/collarbell1.ogg'=1,'modular_zubbers/sound/misc/collarbell2.ogg'=1), 50, 100, 8)
+
+/obj/item/clothing/neck/kink_collar/attack_self(mob/user)
+	..()
+	return tagname
+
+/obj/item/clothing/neck/kink_collar/locked/gps
+	name = "tracking collar"
+	desc = "A collar that lets you find your pet anywhere with GPS!"
+	var/datum/component/gps/gps
+	icon_state = "gps"
+	greyscale_config = /datum/greyscale_config/collar/gps
+	greyscale_config_worn = /datum/greyscale_config/collar/gps/worn
+	greyscale_config_inhand_left = /datum/greyscale_config/collar/gps/lefthand
+	greyscale_config_inhand_right = /datum/greyscale_config/collar/gps/righthand
+	greyscale_colors = "#8B96B7#505665"
+	flags_1 = IS_PLAYER_COLORABLE_1
+	unique_reskin = null
+	resistance_flags = FIRE_PROOF
+
+/obj/item/clothing/neck/kink_collar/locked/gps/Initialize(mapload)
+	. = ..()
+	atom_storage.click_alt_open = FALSE
+	gps = AddComponent(/datum/component/gps, name)
+	register_context()
+	update_icon(UPDATE_OVERLAYS)
+	RegisterSignal(src, COMSIG_NAME_CHANGED, PROC_REF(on_update_name))
+
+/obj/item/clothing/neck/kink_collar/locked/gps/proc/on_update_name()
+	SIGNAL_HANDLER
+	gps.gpstag = name
+
+/obj/item/clothing/neck/kink_collar/locked/gps/attack_self(mob/user)
+	. = ..()
+	gps.gpstag = .
```

Comment:
you could probably just do `gps.gpstag = tagname` instead of overriding parent behaviour.